files/dx_sync_directory.py

```diff
@@ -460,10 +460,10 @@ def upload_tar_files(log, args):
                     opts += '--verbose '
 
                 ua_command = "ua --project %s --folder %s --do-not-compress --wait-on-close --progress %s %s --auth-token %s --chunk-size 25M" % (tar_destination_project, tar_destination_folder, opts, tar_file, args.auth_token)
-                print >> sys.stderr, ua_command
+                print(ua_command, file=sys.stderr)
                 try:
-                    dx_file_id = subprocess.check_output(ua_command, shell=True)
-                    dx_file_id = dx_file_id.strip()
+                    ua_process = subprocess.run(ua_command, shell=True, check=True, stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
Is there an advantage of using run instead of check_output?

By the way in python3 you would need to decode it, in order to get the same result as in python2.
`subprocess.check_output(ua_command, shell=True).decode('utf-8')`

Reply:
the check_output returned bytes instead of string in python3 which caused several issues in the code. Instead check_output I decided to use what was recommended here for python 3.5+ https://queirozf.com/entries/python-3-subprocess-examples#run-command-and-capture-output (see section Run command and capture output)

the decode didn't work well as in other parts of the code it was complaining that it requires bytes instead of string.

Reply:
makes sense, thanks.